Library/Homebrew/formulary.rb

```diff
@@ -260,6 +260,43 @@ def load_file(flags:)
     end
   end
 
+  # Loads a formula from GitHub Packages.
+  class GitHubPackagesLoader < FormulaLoader
```

Comment:
Seems like this should be related to or share code with `FromUrlLoader`?

Reply:
It actually shares more code with `Formulary::BottleLoader`, which is where it's cribbed from.

Reply:
They'll diverge more though as support for reading the image index and manifest is added.

Reply:
There may still be opportunity to factor out common code though.

Reply:
Can it share the code now rather than copy/paste/diverge later?